File: BackEnd/java/ZaloMini/src/main/java/com/devmam/zalomini/entities/GroupInvitation.java
package com.devmam.zalomini.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.Instant;
import java.util.Date;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity(name = "GroupInvitation")
@Table(name = "group_invitations")
public class GroupInvitation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "group_id", nullable = false)
    private ChatGroup group;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "inviter_id", nullable = false)
    private User inviter;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "invitee_id", nullable = false)
    private User invitee;

    @ColumnDefault("'pending'")
    @Lob
    @Column(name = "status")
    private String status;

    @Lob
    @Column(name = "message")
    private String message;

    @Column(name = "expires_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date expiresAt;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @Column(name = "responded_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date respondedAt;

}

File: BackEnd/Zalo_mini/Zalo_mini/Models/group_invitation.cs
using System;
using System.Collections.Generic;

namespace Zalo_mini.Models
{
    public partial class group_invitation
    {
        public long id { get; set; }
        public long group_id { get; set; }
        public long inviter_id { get; set; }
        public long invitee_id { get; set; }
        public string? status { get; set; }
        public string? message { get; set; }
        public DateTime? expires_at { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? responded_at { get; set; }

        public virtual chat_group group { get; set; } = null!;
        public virtual user invitee { get; set; } = null!;
        public virtual user inviter { get; set; } = null!;
    }
}
